### PR TITLE
Union pooler cpp network compatiblity

### PR DESCRIPTION
**Pull request by jaredweiss**
Various updates to the PoolingRegion 

Specifically deal with sending serialized parameters through Network API

### CODE REVIEW

**Comment by jaredweiss** [comment]
@scottpurdy Please review

File: union_pooling/union_pooling/PoolingRegion.py [hunk 1]
```diff
@@ -289,8 +312,10 @@ def getBaseSpec(cls):
           regionLevel=True,
           isDefaultOutput=True),
       ),
-
       parameters=dict(),
+      commands=dict(
+        reset=dict(description='Reset the union pooler.'),
```
Comment (scottpurdy) [inline union_pooling/union_pooling/PoolingRegion.py]: double quotes

File: union_pooling/union_pooling/PoolingRegion.py [hunk 2]
```diff
@@ -252,6 +270,11 @@ def compute(self, inputs, outputs):
     outputs["mostActiveCells"][mostActiveCellsIndices] = 1
 
 
+  def reset(self):
+    """ Reset the state of the Union Pooler """
+    if self._pooler is not None:
+      self._pooler.reset()
+
```
Comment (scottpurdy) [inline union_pooling/union_pooling/PoolingRegion.py]: extra blank line here

File: union_pooling/union_pooling/PoolingRegion.py [hunk 3]
```diff
@@ -228,13 +245,14 @@ def initialize(self, inputs, outputs):
 
     # Retrieve the necessary extra arguments that were handled automatically
     autoArgs = {name: getattr(self, name) for name in self._poolerArgNames}
-    autoArgs["inputDimensions"] = [self._inputWidth]
-    autoArgs["columnDimensions"] = [self._columnCount]
-    autoArgs["potentialRadius"] = self._inputWidth
+    autoArgs["spatialPoolerParams"] = yaml.load(autoArgs["spatialPoolerParams"])
+    autoArgs["spatialPoolerParams"]["inputDimensions"] = [self._inputWidth]
+    autoArgs["spatialPoolerParams"]["columnDimensions"] = [self._columnCount]
+    #autoArgs["spatialPoolerParams"]["potentialRadius"] = self._inputWidth
```
Comment (scottpurdy) [inline union_pooling/union_pooling/PoolingRegion.py]: remove? uncomment?

File: union_pooling/union_pooling/PoolingRegion.py [hunk 4]
```diff
@@ -164,6 +167,21 @@ def getConstraints(arg):
...
+                  "passed during initialization to the spatial pooler. There "
+                  "is currently no way to interact with these parameters after "
+                  "initialization (i.e. the getParameter and setParameter "
+                  "methods have no affect on the SP paramaters, and calls to "
+                  "these methods through the C++ implementation of the network "
+                  "API will likely result in failures.",
+      accessMode="ReadWrite",
+      dataType="Byte",
+      count=0,
+      constraints="")
```
Comment (scottpurdy) [inline union_pooling/union_pooling/PoolingRegion.py]: don't need this anymore!

**Comment by scottpurdy** [comment]
I'd say let's just drop kwargs

**Comment by scottpurdy** [comment]
done with review

**Comment by subutai** [comment]
Is this PR still valid? It needs to be updated to the new structure before we can merge

**Comment by scottpurdy** [comment]
We can create a new PR when there's an updated version.

**Comment by scottpurdy** [comment]
@chetan51 fyi

**Pull request closed without merging**